### ISSUE TITLE
Installing system76 driver on ubuntu when wifi is not functioning

### ISSUE THREAD

**Issue by phs** [system76/docs#311]
On [this page](https://support.system76.com/articles/install-ubuntu/), after performing a clean ubuntu install we are instructed to install the system76 repository and drivers via apt.

However my galago pro's wifi seems to be unavailable in this clean install and hence I can't yet use apt (at least, not directly without first enabling some other internet connection.)

I assume these lines in my `dmesg` are related.  If my situation is typical of a clean install on this model (and if the solution is in fact found within the drivers we're trying to get) then it seems we may need new instructions to work around this chicken & egg issue.

```
$ sudo dmesg | grep iwlwifi
[  265.288061] iwlwifi 0000:00:14.3: TLV_FW_FSEQ_VERSION: FSEQ Version: 43.2.23.17
[  265.288064] iwlwifi 0000:00:14.3: Found debug destination: EXTERNAL_DRAM
[  265.288065] iwlwifi 0000:00:14.3: Found debug configuration: 0
[  265.288271] iwlwifi 0000:00:14.3: loaded firmware version 48.13675109.0 op_mode iwlmvm
[  265.362405] iwlwifi 0000:00:14.3: Detected Intel(R) Wireless-AC 9560, REV=0x354
[  265.369192] iwlwifi 0000:00:14.3: Applying debug destination EXTERNAL_DRAM
[  265.369589] iwlwifi 0000:00:14.3: Allocated 0x00400000 bytes for firmware monitor.
[  266.389990] iwlwifi 0000:00:14.3: Collecting data: trigger 15 fired.
[  266.390065] iwlwifi 0000:00:14.3: Start IWL Error Log Dump:
[  266.390067] iwlwifi 0000:00:14.3: Status: 0x00000000, count: -1987606070
[  266.390068] iwlwifi 0000:00:14.3: Loaded firmware version: 48.13675109.0
[  266.390069] iwlwifi 0000:00:14.3: 0x2F9C089A | ADVANCED_SYSASSERT          
[  266.390069] iwlwifi 0000:00:14.3: 0x256B5677 | trm_hw_status0
[  266.390070] iwlwifi 0000:00:14.3: 0xD27F3757 | trm_hw_status1
[  266.390071] iwlwifi 0000:00:14.3: 0x6ABDC029 | branchlink2
[  266.390072] iwlwifi 0000:00:14.3: 0xC096312C | interruptlink1
[  266.390072] iwlwifi 0000:00:14.3: 0x3312DB99 | interruptlink2
[  266.390073] iwlwifi 0000:00:14.3: 0x87DFD4E4 | data1
[  266.390074] iwlwifi 0000:00:14.3: 0x08FDA76F | data2
[  266.390074] iwlwifi 0000:00:14.3: 0x55A0507D | data3
[  266.390075] iwlwifi 0000:00:14.3: 0x4890E468 | beacon time
[  266.390076] iwlwifi 0000:00:14.3: 0x0387AE8D | tsf low
[  266.390077] iwlwifi 0000:00:14.3: 0xA3719EC9 | tsf hi
[  266.390077] iwlwifi 0000:00:14.3: 0x78635F59 | time gp1
[  266.390078] iwlwifi 0000:00:14.3: 0x61797522 | time gp2
[  266.390079] iwlwifi 0000:00:14.3: 0x4B20D912 | uCode revision type
[  266.390079] iwlwifi 0000:00:14.3: 0xDD03A702 | uCode version major
[  266.390080] iwlwifi 0000:00:14.3: 0x1A25C79A | uCode version minor
[  266.390081] iwlwifi 0000:00:14.3: 0xC861B4D3 | hw version
[  266.390081] iwlwifi 0000:00:14.3: 0x4C55C566 | board version
[  266.390082] iwlwifi 0000:00:14.3: 0xB3C1EA76 | hcmd
[  266.390083] iwlwifi 0000:00:14.3: 0xAF99C2F3 | isr0
[  266.390083] iwlwifi 0000:00:14.3: 0xB87BF372 | isr1
[  266.390085] iwlwifi 0000:00:14.3: 0x4EA41D3F | isr2
[  266.390085] iwlwifi 0000:00:14.3: 0x3A94D381 | isr3
[  266.390086] iwlwifi 0000:00:14.3: 0xE763FA53 | isr4
[  266.390087] iwlwifi 0000:00:14.3: 0x0CD884CE | last cmd Id
[  266.390087] iwlwifi 0000:00:14.3: 0xDA73871B | wait_event
[  266.390088] iwlwifi 0000:00:14.3: 0x2363DDCF | l2p_control
[  266.390089] iwlwifi 0000:00:14.3: 0x4CB7AE90 | l2p_duration
[  266.390089] iwlwifi 0000:00:14.3: 0x28C218B7 | l2p_mhvalid
[  266.390090] iwlwifi 0000:00:14.3: 0x89425401 | l2p_addr_match
[  266.390091] iwlwifi 0000:00:14.3: 0x9111B146 | lmpm_pmg_sel
[  266.390092] iwlwifi 0000:00:14.3: 0x6063D16E | timestamp
[  266.390092] iwlwifi 0000:00:14.3: 0xBAB682D7 | flow_handler
[  266.390122] iwlwifi 0000:00:14.3: Start IWL Error Log Dump:
[  266.390123] iwlwifi 0000:00:14.3: Status: 0x00000000, count: 7
[  266.390124] iwlwifi 0000:00:14.3: 0x201013F1 | ADVANCED_SYSASSERT
[  266.390125] iwlwifi 0000:00:14.3: 0x00000000 | umac branchlink1
[  266.390125] iwlwifi 0000:00:14.3: 0xC008CF5C | umac branchlink2
[  266.390126] iwlwifi 0000:00:14.3: 0x00000000 | umac interruptlink1
[  266.390127] iwlwifi 0000:00:14.3: 0x00000000 | umac interruptlink2
[  266.390128] iwlwifi 0000:00:14.3: 0x00000003 | umac data1
[  266.390128] iwlwifi 0000:00:14.3: 0x20000302 | umac data2
[  266.390129] iwlwifi 0000:00:14.3: 0x01300202 | umac data3
[  266.390130] iwlwifi 0000:00:14.3: 0x00000030 | umac major
[  266.390130] iwlwifi 0000:00:14.3: 0x13675109 | umac minor
[  266.390131] iwlwifi 0000:00:14.3: 0x00005CF2 | frame pointer
[  266.390132] iwlwifi 0000:00:14.3: 0xC0887F58 | stack pointer
[  266.390132] iwlwifi 0000:00:14.3: 0x00000000 | last host cmd
[  266.390133] iwlwifi 0000:00:14.3: 0x00000000 | isr status reg
[  266.390147] iwlwifi 0000:00:14.3: Fseq Registers:
[  266.390149] iwlwifi 0000:00:14.3: 0x00000003 | FSEQ_ERROR_CODE
[  266.390151] iwlwifi 0000:00:14.3: 0x00000000 | FSEQ_TOP_INIT_VERSION
[  266.390153] iwlwifi 0000:00:14.3: 0x993C007E | FSEQ_CNVIO_INIT_VERSION
[  266.390156] iwlwifi 0000:00:14.3: 0x0000A384 | FSEQ_OTP_VERSION
[  266.390158] iwlwifi 0000:00:14.3: 0xF7E26237 | FSEQ_TOP_CONTENT_VERSION
[  266.390160] iwlwifi 0000:00:14.3: 0x70E09BCC | FSEQ_ALIVE_TOKEN
[  266.390163] iwlwifi 0000:00:14.3: 0x3B1F3821 | FSEQ_CNVI_ID
[  266.390165] iwlwifi 0000:00:14.3: 0xEB1E049F | FSEQ_CNVR_ID
[  266.390167] iwlwifi 0000:00:14.3: 0x20000302 | CNVI_AUX_MISC_CHIP
[  266.390171] iwlwifi 0000:00:14.3: 0x01300202 | CNVR_AUX_MISC_CHIP
[  266.390176] iwlwifi 0000:00:14.3: 0x0000485B | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[  266.390210] iwlwifi 0000:00:14.3: 0xA5A5A5A2 | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[  266.390270] iwlwifi 0000:00:14.3: SecBoot CPU1 Status: 0x5c63, CPU2 Status: 0x3
[  266.390271] iwlwifi 0000:00:14.3: Failed to start RT ucode: -110
[  266.390273] iwlwifi 0000:00:14.3: Firmware not running - cannot dump error
[  266.402044] iwlwifi 0000:00:14.3: Failed to run INIT ucode: -110
```

**Comment by phs** [comment]
Indeed, connecting other ethernet and installing the system76 apt repository and driver allows the wifi to begin working.

**Comment by phs** [comment]
Respectfully, this is not yet resolved.

I would expect at least a warning at the top of the page calling out the specific situations in which an alternate internet connection is needed, similar to the one for oryx pro touchpad.

**Comment by ahoneybun** [comment]
I'm sorry about that as it seems the Github mobile application didn't let me leave a note. Those messages are usually from a hardware issue though it could be that you have a galp4 which needs at least a 5.3 kernel to have working wireless though you didn't include any other information other then dmesg about the iwlwifi driver like what OS version you installed and what model of the Galago Pro which would be helpful.

**Comment by phs** [comment]
Sure thing.

> though it could be that you have a galp4

Yes, it's a galp4 per the bottom label.

> which needs at least a 5.3 kernel to have working wireless

The ubuntu iso I ran was a stock 18.04 LTS desktop downloaded a few days ago.  If the kernel it ships isn't new enough, then it seems we can expect other galp4 users to encounter the same problem.

I'm happy to share other details of my scenario if they would be useful.

**Comment by ahoneybun** [comment]
It would also happen to the darp6 as well due to the 10th Gen CPU's though we haven't seen any issues with the default kernel that Ubuntu ships with when it comes to this package:

linux-generic-hwe-18.04	5.3.0.28.96

Which is shipping with the 18.04.4 ISO.

**Comment by phs** [comment]
I've upgraded the kernel since installing (I did that at the same time as installing the system76 driver, so really as far as I know either could have been the fix.)

My /etc/issue does read 18.04.4

Phil Smith (phs)
https://twitter.com/philhsmith
https://github.com/phs

**Comment by ahoneybun** [comment]
I wanted to be sure that the issue was adjusted before closing this issue.

**Comment by withinwheels** [comment]
Thanks for checking in.  I don't know if the issue was specific somehow to me or not, I have not re-attempted the install process since.

**Comment by phs** [comment]
Thanks for checking in.  I don't know if the issue was specific somehow to me or not, I have not re-attempted the install process since.  It seems like as a fresh install I would be similar to others, though..

**Comment by ahoneybun** [comment]
We could add a note at the top of the article that a 5.8 kernel needs to be installed (which is installed with the system76-driver) for newer systems to have working Wi-Fi.

**Comment by storyaddict** [comment]
@ahoneybun why is this labelled as invalid? Also the kernel/Intel gen info should still be added if it still applies.

**Comment by ahoneybun** [comment]
Since we aren't shipped any distros with a kernel older then 5.8 I believe it is invalid.

**Comment by Ampersandstorm** [comment]
@ahoneybun @storyaddict Can this be closed?
